docs/upload.rst

```diff
@@ -314,10 +330,12 @@ Upload: /upload/
    :statuscode 403: your auth token is invalid and you need to get a new one
    :statuscode 413: your upload is too large; split it into smaller files or switch to
        upload by download url
-   :statuscode 429: sleep for a bit and retry
-   :statuscode 500: sleep for a bit and retry; if retrying doesn't work, then please
-       file a bug report
-   :statuscode 503: sleep for a bit and retry
+   :statuscode 429: your request has been rate-limited; sleep for a bit and retry
+   :statuscode 500: there's an error with the server; sleep for a bit and
+       retry; if retrying doesn't work, then please file a bug report
+   :statuscode 502: sleep for a bit and retry
+   :statuscode 504: the request is taking too long to complete; sleep for a bit
+       and retry
```

Comment:
I also found in the [Tecken prod logs over the last week 29 occurrences (0.3% of all response codes) of a 499 response code](https://console.cloud.google.com/logs/query;query=logName%3D%2528%22projects%2Fmoz-fx-tecken-prod-da83%2Flogs%2Ftecken_nginx_access%22%20OR%20%22projects%2Fmoz-fx-tecken-prod-da83%2Flogs%2Ftecken_nginx_error%22%2529%0AjsonPayload.request%3D%22POST%20%2Fupload%2F%20HTTP%2F1.1%22%0Ajson_payload.code%3D%22499%22;storageScope=storage,projects%2Fmoz-fx-tecken-prod-da83%2Flocations%2Fglobal%2Fbuckets%2F_Default%2Fviews%2F_AllLogs,projects%2Fmoz-fx-tecken-prod-da83%2Flocations%2Fglobal%2Fbuckets%2F_Default%2Fviews%2F_Default,projects%2Fmoz-fx-tecken-prod-da83%2Flocations%2Fglobal%2Fbuckets%2F_Required%2Fviews%2F_AllLogs;summaryFields=:false:32:beginning;lfeCustomFields=jsonPayload%252Fcode;cursorTimestamp=2024-07-13T09:47:43Z;duration=P7D?project=moz-fx-tecken-prod-da83). Interestingly, this is a non-standard code (not in MDN), but it appears to be a code used by nginx to indicate the client closed the connection before the server could send a response. I couldn't find this in nginx's docs, but I saw a bunch of articles about it with an organic web search.

Reply:
That's not a response code clients will ever see, though. It's never actually _sent_ by nginx, since the client already closed the connection. It's only recorded in the logs. We don't need to document this since it's not visible from the outside.